Add support for enthalpy variable in nonlinear physics

GiudGiud opened the pull request:
refs https://github.com/idaholab/moose/issues/31043

GiudGiud committed:
Add helpful comments

GiudGiud committed:
Enum the inlet types

GiudGiud committed:
Add enthalpy support in energy flux BC
Add a FP UO to get enthalpy from boundary temperature
refs #31043

GiudGiud committed:
Retrieve the momentum inlet type from the Physics
Error if incompatible with energy inlet type

GiudGiud committed:
Fix a typo on thermal conductivity in Rayleigh number doco

GiudGiud committed:
Add support for enthalpy variable in nonlinear finite volume fluid energy Physics
refs #31043

GiudGiud committed:
Add support for computing T from h in INSFV functor material
refs #31043

GiudGiud committed:
Use the functor material dedicated to linearFV or nonlinearFV in fluid heat transfer physics
refs #31043

GiudGiud committed:
Add test for using enthalpy variable
refs #31043

GiudGiud committed:
Check the FE ICs for ICs defining Fv variables
Add ICs on enthalpy when defining the temperature
refs #30533

GiudGiud committed:
Use FVICs as they might be faster to initialize temperature and enthalpy

GiudGiud committed:
Create energy flux BC in enthalpy mode when solving for enthalpy
refs #31043

GiudGiud committed:
Add a BC to compute a dirichlet on enthalpy from P & T
refs #31043

GiudGiud committed:
Create the new dirichlet BC in the physics when solving for enthalpy
refs #31043

GiudGiud committed:
Set FP even when not solving for enthalpy in flux BCs
refs #31043

GiudGiud committed:
Improve matprop missing output warning message

GiudGiud committed:
Use k / cp for diffusion term

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/3044165/), step Clang format on 82f4009 wanted to post the following:

Your code requires style changes.  

A patch was auto generated and copied [here](https://mooseframework.inl.gov/docs/PRs/31036/clang_format/style.patch)  
You can directly apply the patch by running, in the top level of your repository:

`curl -s https://mooseframework.inl.gov/docs/PRs/31036/clang_format/style.patch | git apply -v`  

Alternatively, with your repository up to date and in the top level of your repository:

`git clang-format 97d07485be4caabe405a33494c2a937208d1ff81`

GiudGiud committed:
Check flux inlet pps before adding them

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/3044398/), step Clang format on aa09300 wanted to post the following:

Your code requires style changes.  

A patch was auto generated and copied [here](https://mooseframework.inl.gov/docs/PRs/31036/clang_format/style.patch)  
You can directly apply the patch by running, in the top level of your repository:

`curl -s https://mooseframework.inl.gov/docs/PRs/31036/clang_format/style.patch | git apply -v`  

Alternatively, with your repository up to date and in the top level of your repository:

`git clang-format 97d07485be4caabe405a33494c2a937208d1ff81`

GiudGiud committed:
Make check on BC compatiblity an error by default, with an option to skip it

moosebuild commented:
Job [Documentation](https://civet.inl.gov/job/3048532/), step Docs: sync website on b897067 wanted to post the following:

View the site [here](https://mooseframework.inl.gov/docs/PRs/31036/site)

This comment will be updated on new commits.

GiudGiud committed:
Adjust logic for defining the initial temperature boundary condition:
- dont define BC for enthalpy when not solving for it
- check both enthalpy and T BC for use of the initial_temperature parameter

GiudGiud committed:
Clean up inputs

GiudGiud committed:
Fix logic for fluxBC outflow: use T_fluid variable not T_pp

GiudGiud committed:
Use fv_ic task for WCNSFV / linearFV physics, to avoid adding ICs from Physics before FVICs

moosebuild commented:
Job [Coverage](https://civet.inl.gov/job/3099301/), step Generate coverage on dfc4326 wanted to post the following:


## Framework coverage

<table>
  <thead>
    <tr>
      <th></th>
      <th>ee92e3</th>
      <th colspan=4>#31036 dfc432</th>
    </tr>
    <tr>
      <th></th>
      <th>Total</th>
      <th>Total</th>
      <th>+/-</th>
      <th>New</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td><b>Rate</b></td>
      <td>85.54%</td>
      <td>85.55%</td>
      <td>+0.00%</td>
      <td>37.50%</td>
    </tr>
    <tr>
      <td><b>Hits</b></td>
      <td>116194</td>
      <td>116202</td>
      <td>+8</td>
      <td>3</td>
    </tr>
    <tr>
      <td><b>Misses</b></td>
      <td>19634</td>
      <td>19632</td>
      <td>-2</td>
      <td>5</td>
    </tr>
  </tbody>
</table>

[Diff coverage report](https://mooseframework.inl.gov/docs/PRs/31036/coverage_diff/framework)

[Full coverage report](https://mooseframework.inl.gov/docs/PRs/31036/coverage/framework)


## Modules coverage

### Navier stokes

<table>
  <thead>
    <tr>
      <th></th>
      <th>ee92e3</th>
      <th colspan=4>#31036 dfc432</th>
    </tr>
    <tr>
      <th></th>
      <th>Total</th>
      <th>Total</th>
      <th>+/-</th>
      <th>New</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td><b>Rate</b></td>
      <td>85.40%</td>
      <td>85.41%</td>
      <td>+0.01%</td>
      <td>89.87%</td>
    </tr>
    <tr>
      <td><b>Hits</b></td>
      <td>19760</td>
      <td>19867</td>
      <td>+107</td>
      <td>213</td>
    </tr>
    <tr>
      <td><b>Misses</b></td>
      <td>3378</td>
      <td>3394</td>
      <td>+16</td>
      <td>24</td>
    </tr>
  </tbody>
</table>

[Diff coverage report](https://mooseframework.inl.gov/docs/PRs/31036/coverage_diff/navier_stokes)

[Full coverage report](https://mooseframework.inl.gov/docs/PRs/31036/coverage/navier_stokes)

## Full coverage reports

<details>
<summary>Reports</summary>

- [`framework`](https://mooseframework.inl.gov/docs/PRs/31036/coverage/framework)
- [`chemical_reactions`](https://mooseframework.inl.gov/docs/PRs/31036/coverage/chemical_reactions)
- [`combined`](https://mooseframework.inl.gov/docs/PRs/31036/coverage/combined)
- [`contact`](https://mooseframework.inl.gov/docs/PRs/31036/coverage/contact)
- [`electromagnetics`](https://mooseframework.inl.gov/docs/PRs/31036/coverage/electromagnetics)
- [`external_petsc_solver`](https://mooseframework.inl.gov/docs/PRs/31036/coverage/external_petsc_solver)
- [`fluid_properties`](https://mooseframework.inl.gov/docs/PRs/31036/coverage/fluid_properties)
- [`fsi`](https://mooseframework.inl.gov/docs/PRs/31036/coverage/fsi)
- [`functional_expansion_tools`](https://mooseframework.inl.gov/docs/PRs/31036/coverage/functional_expansion_tools)
- [`geochemistry`](https://mooseframework.inl.gov/docs/PRs/31036/coverage/geochemistry)
- [`heat_transfer`](https://mooseframework.inl.gov/docs/PRs/31036/coverage/heat_transfer)
- [`level_set`](https://mooseframework.inl.gov/docs/PRs/31036/coverage/level_set)
- [`misc`](https://mooseframework.inl.gov/docs/PRs/31036/coverage/misc)
- [`navier_stokes`](https://mooseframework.inl.gov/docs/PRs/31036/coverage/navier_stokes)
- [`optimization`](https://mooseframework.inl.gov/docs/PRs/31036/coverage/optimization)
- [`peridynamics`](https://mooseframework.inl.gov/docs/PRs/31036/coverage/peridynamics)
- [`phase_field`](https://mooseframework.inl.gov/docs/PRs/31036/coverage/phase_field)
- [`porous_flow`](https://mooseframework.inl.gov/docs/PRs/31036/coverage/porous_flow)
- [`ray_tracing`](https://mooseframework.inl.gov/docs/PRs/31036/coverage/ray_tracing)
- [`rdg`](https://mooseframework.inl.gov/docs/PRs/31036/coverage/rdg)
- [`reactor`](https://mooseframework.inl.gov/docs/PRs/31036/coverage/reactor)
- [`richards`](https://mooseframework.inl.gov/docs/PRs/31036/coverage/richards)
- [`scalar_transport`](https://mooseframework.inl.gov/docs/PRs/31036/coverage/scalar_transport)
- [`solid_mechanics`](https://mooseframework.inl.gov/docs/PRs/31036/coverage/solid_mechanics)
- [`solid_properties`](https://mooseframework.inl.gov/docs/PRs/31036/coverage/solid_properties)
- [`stochastic_tools`](https://mooseframework.inl.gov/docs/PRs/31036/coverage/stochastic_tools)
- [`subchannel`](https://mooseframework.inl.gov/docs/PRs/31036/coverage/subchannel)
- [`thermal_hydraulics`](https://mooseframework.inl.gov/docs/PRs/31036/coverage/thermal_hydraulics)
- [`xfem`](https://mooseframework.inl.gov/docs/PRs/31036/coverage/xfem)

</details>

## Warnings

- `framework` new line coverage rate 37.50% is less than the suggested 90.0%
- `navier_stokes` new line coverage rate 89.87% is less than the suggested 90.0%



This comment will be updated on new commits.

GiudGiud committed:
Address test failures:
- SQA requirement de-duplicated
- Add missing header for non unity build

GiudGiud commented on modules/navier_stokes/doc/content/source/postprocessors/RayleighNumber.md:
```diff
@@ -15,7 +15,7 @@ where:
 - $g$ is the magnitude of the gravity
 - $\mu$ is the dynamic viscosity of the fluid
-- $k$ is the thermal diffusivity of the fluid
+- $k$ is the thermal conductivity of the fluid
```
```suggestion
- $\alpha$ is the thermal diffusivity of the fluid
```

GiudGiud committed:
Address Alex's review

- static cast
- fixed near failure on rayleigh doco
- parse to real an output when checking if parsedToReal

Co-authored-by: Alex Lindsay <alexander.lindsay@inl.gov>

lindsayad commented on modules/navier_stokes/src/functormaterials/INSFVEnthalpyFunctorMaterial.C:
```diff
@@ -100,25 +106,33 @@ INSFVEnthalpyFunctorMaterial::INSFVEnthalpyFunctorMaterial(const InputParameters
           Real h, dh_dp, dh_dT;
-          _fp->h_from_p_T((*_pressure)(r, t).value(), _temperature(r, t).value(), h, dh_dp, dh_dT);
-          return dh_dT * _temperature.dot(r, t) + dh_dp * _pressure->dot(r, t);
+          _fp->h_from_p_T(
+              (*_pressure)(r, t).value(), (*_temperature)(r, t).value(), h, dh_dp, dh_dT);
```
I'm not suggesting you make this a `Real` functor. Is there no 5-arg AD overload for `h_from_p_T`?

GiudGiud replied:
It’s declared so people can override it but in practice it’s never împlemented. 

That shape exists for rho for some select fluid properties that got attention for NS purposes

lindsayad replied:
The use of `.value()` predates the PR. I haven't added "requested changes" for my part. So it's not a demand of mine that this has to be changed. 

But I do think it would be worth adding more `ADReal` 5 argument overloads. You should be able to simply add a template argument to existing `Real` implementations and call them helpers and use them for both `Real` and `ADReal` overloads right?

GiudGiud replied:
changed

The tradeoff is that every single user of transient + h-variable + newtonFV will need to set allow_imperfect_jacobians in their FluidProperties object. Because there are 0 implementations of ADReal 5-arg overload for h in the module.

lindsayad replied:
Thanks. I think it's better that a user will have explicitly acknowledged that the jacobians will be imperfect than for them to be quietly imperfect (note I'm not saying that imperfect has real consequences ... convergence can still be good if the Jacobian is imperfect. Imperfect can still be great)

grmnptr reviewed (changes requested):
Nice!
Could we add a Newton-equivalent of this test?
https://github.com/idaholab/moose/blob/next/modules/navier_stokes/test/tests/finite_volume/wcns/enthalpy_equation/1d_test_h.i
To make sure we are getting the analytic solution back.

GiudGiud commented on modules/navier_stokes/src/fvbcs/WCNSFVEnergyFluxBC.C:
```diff
@@ -99,3 +113,33 @@ WCNSFVEnergyFluxBC::isInflow() const
+                             const Moose::StateArg & state,
+                             const bool inflow) const
+{
+  // Outlet, use interior values
+  if (!inflow)
+  {
+    if (_h_fluid)
+      return (*_h_fluid)(loc_arg, state);
+    else if (_fluid)
+      return _fluid->h_from_p_T((*_pressure)(loc_arg, state), _temperature(loc_arg, state));
+    else if (_temperature_pp)
+      return (*_cp)(loc_arg, state) * _temperature(loc_arg, state);
+  }
+  else
+  {
+    if (_temperature_pp)
+    {
+      // Preferable to use the fluid property if we know it
+      if (_fluid)
+        return _fluid->h_from_p_T((*_pressure)(loc_arg, state), (*_temperature_pp));
+      else if (_cp)
+        return (*_cp)(loc_arg, state) * (*_temperature_pp);
+    }
+  }
```
not really. Ideally we create these BCs through the Physics at this point. So both options are acceptable.

But one object per case is a lot of objects, documentation, etc.

GiudGiud committed:
Add a test showing the 1d analytical energy conservation for nonlinearFV using enthalpy variables

GiudGiud committed:
Dont neglect any derivative in INSFV EnthalpyFunctorMaterial enthalpy call
Every single user will need to 'allow_imperfect_jacobian' as no one implemented h_from_p_T(AD pressure, AD temp, AD h, AD dh/dT, AD dh/dp)

GiudGiud committed:
Add an option to avoid using cp * T with fluid properties

GiudGiud committed:
Add some overrides for ADreal T_from_p_h

GiudGiud commented:
Pronghorn patch:
https://github.inl.gov/ncrc/pronghorn/pull/1306

lindsayad commented on modules/navier_stokes/src/functormaterials/INSFVEnthalpyFunctorMaterial.C:
```diff
@@ -100,25 +106,32 @@ INSFVEnthalpyFunctorMaterial::INSFVEnthalpyFunctorMaterial(const InputParameters
-          Real h, dh_dp, dh_dT;
-          _fp->h_from_p_T((*_pressure)(r, t).value(), _temperature(r, t).value(), h, dh_dp, dh_dT);
-          return dh_dT * _temperature.dot(r, t) + dh_dp * _pressure->dot(r, t);
+          ADReal h, dh_dp, dh_dT;
+          _fp->h_from_p_T((*_pressure)(r, t), (*_temperature)(r, t), h, dh_dp, dh_dT);
```
❤️

lindsayad commented on modules/navier_stokes/src/physics/WCNSFVFluidHeatTransferPhysics.C:
```diff
@@ -403,16 +482,67 @@ WCNSFVFluidHeatTransferPhysics::addEnergyWallBC()
+void
+WCNSFVFluidHeatTransferPhysics::addMaterials()
+{
+  if (!_has_energy_equation)
```
hopefully we can remove this boolean soon?

GiudGiud replied:
yeah at some point. I would have to transition the rest of the inputs.

lindsayad commented on modules/navier_stokes/src/physics/WCNSFVFluidHeatTransferPhysics.C:
```diff
@@ -403,16 +482,67 @@ WCNSFVFluidHeatTransferPhysics::addEnergyWallBC()
+  if (!_has_energy_equation)
+    return;
+
+  // Note that this material choice would not work for Newton-INSFV + solve_for_enthalpy
```
An explanation of why not would be appreciated

lindsayad replied:
Especially because the material name is `INSFVEnthalpyFunctorMaterial` I would expect full support for (Newton-)INSFV

GiudGiud replied:
It's to compute the enthalpy from temperature, which one should not need to do when solving for enthalpy

lindsayad commented on modules/navier_stokes/test/tests/finite_volume/wcns/enthalpy_equation/1d_test_h_fp-nonlinearFV.i:
```diff
@@ -0,0 +1,135 @@
+        coupled_flow_physics = flow
+
+        solve_for_enthalpy = true
+        # There are no fluid temperature (auxiliary) variable when solving for enthalpy
```
```suggestion
        # There is no fluid temperature (auxiliary) variable when solving for enthalpy
```

lindsayad commented on modules/navier_stokes/test/tests/finite_volume/wcns/enthalpy_equation/1d_test_h_fp-nonlinearFV.i:
```diff
@@ -0,0 +1,135 @@
+        initial_enthalpy = '${fparse 800 * 240}'
+
+        energy_inlet_types = 'flux-velocity'
+        # specifies inlet temperature, not inlet enthalpy
+        energy_inlet_functors = '${T_in}'
```
it troubles me that temperature is used as an energy

GiudGiud replied:
it is odd but not sure how to do better with a single "energy_inlet_functors" parameter that maps to a single "energy_inlet_boundaries" parameter. I really dont want to specify an inlet because that's too much work to obtain. i think my average workflow if I had to use enthalpy would be to pull up the source code for the fluid properties and compute the enthalpy I want by hand

We would need to add maybe "energy_temperature_boundaries", but here the flux boundary conditions being used are not temperature BCs. I think the code could guess that the temperature is a "flux-inlet" with a fixed temperature from seeing the boundary appear in the two parameters.

lindsayad commented on modules/navier_stokes/test/tests/finite_volume/wcns/enthalpy_equation/1d_test_h_fp-nonlinearFV.i:
```diff
@@ -0,0 +1,135 @@
+
+[Postprocessors]
+  [T_out_sim]
+    type = ElementalVariableValue
```
I think I'd prefer `PointValue` over `ElementalVariableValue`. The logic in the latter class actually seems pretty wack to me ... just averaging the values at quadrature points with no regard for the values of JxW at those points

lindsayad replied:
I've opened https://github.com/idaholab/moose/issues/31667 with respect to `ElementalVariableValue`

GiudGiud replied:
sounds good. we should be able to use either here, since it's FV the good or bad averaging give the same result

lindsayad commented on modules/navier_stokes/test/tests/finite_volume/wcns/enthalpy_equation/tests:
```diff
@@ -16,11 +16,19 @@
+      type = CSVDiff
+      input = 1d_test_h_fp-nonlinearFV.i
+      csvdiff = 1d_test_h_fp-nonlinearFV_out.csv
+      detail = 'using an enthalpy function defined in fluid properties and the nonlinear finite volume discretization'
```
```suggestion
      detail = 'using an enthalpy function defined in fluid properties and the nonlinear finite volume implementation'
```

lindsayad commented on modules/navier_stokes/test/tests/finite_volume/wcns/enthalpy_equation/tests:
```diff
@@ -40,4 +48,26 @@
+    cli_args = "Physics/NavierStokes/Flow/flow/momentum_inlet_types='flux-velocity'
+                Physics/NavierStokes/Flow/flow/flux_inlet_pps='${bulk_u}'"
+    csvdiff = 'enthalpy_equation-physics-nonlinearFV_out.csv'
+    requirement = 'The system shall be able to solve a two-dimensional enthalpy conservation equation using fluid properties for the definition of enthalpy and the shorthand physics syntax, using the nonlinear finite volume discretization and flux boundary conditions in the momentum equation.'
```
```suggestion
    requirement = 'The system shall be able to solve a two-dimensional enthalpy conservation equation using fluid properties for the definition of enthalpy and the shorthand physics syntax, using the nonlinear finite volume implementation and flux boundary conditions in the momentum equation.'
```

lindsayad commented on modules/navier_stokes/test/tests/finite_volume/wcns/enthalpy_equation/tests:
```diff
@@ -40,4 +48,26 @@
+    # we expect some difference between dirichlet and flux BCs for a non-constant cp non-constant rho fluid
+    rel_err = 2e-3
+    prereq = '2d-fp-physics-flux-vel-inlet'
+    requirement = 'The system shall be able to solve a two-dimensional enthalpy conservation equation using fluid properties for the definition of enthalpy and the shorthand physics syntax, using the nonlinear finite volume discretization and Dirichlet boundary conditons on velocity.'
```
```suggestion
    requirement = 'The system shall be able to solve a two-dimensional enthalpy conservation equation using fluid properties for the definition of enthalpy and the shorthand physics syntax, using the nonlinear finite volume implementation and Dirichlet boundary conditions on velocity.'
```

GiudGiud committed:
Address review

- typos and discretization -> implementation
- add comment why we dont use enthalpy functor material

Co-authored-by: Alex Lindsay <alexander.lindsay@inl.gov>

GiudGiud commented on modules/navier_stokes/test/tests/finite_volume/wcns/enthalpy_equation/1d_test_h_fp-nonlinearFV.i:
```diff
@@ -0,0 +1,135 @@
+    type = ElementalVariableValue
+    variable = T_out
+    elementid = ${fparse nx-1}
+  []
```
note that I sent you an input file for the non-FP case that got the expected result, for the report, but it was a little convoluted iirc

grmnptr commented on modules/navier_stokes/src/functormaterials/INSFVEnthalpyFunctorMaterial.C:
```diff
@@ -100,25 +106,32 @@ INSFVEnthalpyFunctorMaterial::INSFVEnthalpyFunctorMaterial(const InputParameters
                                [this](const auto & r, const auto & t) { return _h->dot(r, t); });
+
+    // Note that we don't pick the fluid temperature name
+    addFunctorProperty<ADReal>(NS::T_fluid,
```
Would be nice to have dicumentation that this object is creating this functor under the hood.

GiudGiud committed:
Address Peter's review

- typo in parameter description
- add documentation on functor creation

Co-authored-by: Peter German <31662443+grmnptr@users.noreply.github.com>

grmnptr reviewed (commented):
As long as Alex's concerns are addressed I think this is good to go!

GiudGiud committed:
Allow renumbering in the test by using ptvalue for PPing

moosebuild commented:
Job [Test](https://civet.inl.gov/job/3320608/), step Results summary on 44c8729 wanted to post the following:

## Framework test summary

### Removed tests
None
### Added tests
None
### Run time changes
| Test                                                                                         |   Base (s) |   Head (s) | +/-     |
|----------------------------------------------------------------------------------------------|------------|------------|---------|
| `vectorpostprocessors/element_value_sampler.element_value_sampler/fv`                        |       0.85 |       1.46 | +71.31% |
| `convergence/reference_residual_convergence.wildcard`                                        |       0.83 |       1.41 | +69.93% |
| `ics/functor_ic.functor/from_fmat`                                                           |       0.86 |       1.45 | +69.61% |
| `mesh/add_sideset_ids.add_sideset_names`                                                     |       0.92 |       1.51 | +65.30% |
| `transfers/multiapp_reporter_transfer/dist_vector.vec_rep/reporter_transfer_limit_max_procs` |       1    |       1.63 | +63.14% |
| `controls/error.multiple_params_found_warning`                                               |       1.07 |       1.73 | +61.37% |
| `transfers/general_field/nearest_node/between_siblings.from_sub_to_sub/nodal_nodal`          |       0.99 |       1.59 | +61.29% |
| `userobjects/domain-user-object.domain_uo`                                                   |       0.85 |       1.36 | +61.17% |
| `postprocessors/side_extreme_value.g/aux_nodal`                                              |       0.87 |       1.39 | +59.39% |
| `usability.command-line/empty`                                                               |       0.82 |       1.3  | +58.21% |
| `meshgenerators/batch_mesh_generation_action.double_scalar_cp`                               |       0.77 |       1.22 | +57.93% |
| `time_integrators/convergence.explicit_ralston/level1`                                       |       0.92 |       1.45 | +57.11% |
| `misc/deprecation.expired_error`                                                             |       1.23 |       1.93 | +56.88% |
| `mfem/submeshes.MFEMDomainSubMesh`                                                           |       0.86 |       1.35 | +56.52% |
| `functors/previous-nl-it.functors/lag`                                                       |       0.87 |       1.35 | +56.25% |
| `variables/fe_hermite.group/test_hermite_3_1d`                                               |       0.85 |       1.33 | +56.06% |
| `bcs/ad_bcs.vector_test`                                                                     |       0.89 |       1.39 | +56.01% |
| `outputs/debug.show_top_residuals_nonlinear_only`                                            |       0.83 |       1.29 | +55.90% |
| `meshgenerators/batch_mesh_generation_action.fixed_real_vector_value_mixed`                  |       0.78 |       1.21 | +55.21% |
| `restart/restart_transient_from_steady.restart_trans_from_steady_with_2subs`                 |       1.04 |       1.6  | +54.69% |
| `dirackernels/reporter_point_source.reporterPointSource/2dConstantReporter`                  |       1.08 |       1.67 | +54.63% |
| `meshgenerators/lower_d_block_generator.mesh/names`                                          |       0.92 |       1.42 | +54.53% |
| `transfers/general_field/user_object/nearest_position.2d_overlay/nearest_position`           |       1.11 |       1.72 | +53.95% |
| `mfem/kernels.MFEMGradDiv`                                                                   |       0.88 |       1.35 | +53.64% |
| `outputs/checkpoint.default/wall_time_interval`                                              |       1.19 |       1.82 | +52.94% |
| `actioncomponents.combining_meshes`                                                          |       0.85 |       1.29 | +52.70% |
| `mfem/ics.MFEMTransientScalarIC`                                                             |       0.87 |       1.33 | +52.60% |
| `transfers/general_field/user_object/regular.2d_overlay/projection_needed_receiving`         |       1.15 |       1.76 | +52.37% |
| `kernels/ad_reaction.ad_coef_reaction_jac`                                                   |       0.88 |       1.33 | +52.02% |
| `preconditioners/auto_smp.petsc_pre_description`                                             |       0.89 |       1.35 | +51.83% |
| `kernels/2d_diffusion.testbodyforce`                                                         |       0.74 |       1.13 | +51.46% |
| `transfers/multiapp_projection_transfer.fixed_meshes`                                        |       0.96 |       1.46 | +51.29% |
| `scaling/ignore-variables.ignore`                                                            |       1.7  |       2.56 | +50.90% |
| `outputs/console.file_system_information_2procs_head`                                        |       0.84 |       1.26 | +50.33% |
## Modules test summary

### Removed tests
None
### Added tests
| Test                                                                                                      |   Time (s) |
|-----------------------------------------------------------------------------------------------------------|------------|
| `navier_stokes/test:finite_volume/wcns/enthalpy_equation.1d/analytical_outlet_temperature_fp_nonlinearFV` |       2.99 |
| `navier_stokes/test:finite_volume/wcns/enthalpy_equation.2d-fp-physics-dirichlet-vel-and-T-inlet`         |       1.8  |
| `navier_stokes/test:finite_volume/wcns/enthalpy_equation.2d-fp-physics-flux-vel-inlet`                    |       1.58 |
### Run time changes
| Test                                                               |   Base (s) |   Head (s) | +/-     |
|--------------------------------------------------------------------|------------|------------|---------|
| `thermal_hydraulics/test:controls/get_function_value_control.test` |       1.14 |       1.82 | +59.97% |

GiudGiud commented:
Thanks for the reviews!
